client/alloc_runner.go

```diff
@@ -302,7 +309,7 @@ func (r *AllocRunner) Run() {
 	r.taskLock.Lock()
 	for _, task := range tg.Tasks {
 		// Skip tasks that were restored
-		if _, ok := r.tasks[task.Name]; ok {
+		if _, ok := r.taskStatus[task.Name]; ok {
```

Comment:
@dadgar I think it would be clean to not start TaskRunners while restoring them. Restore state could just recreate the state of the structs, and when the client finishes calling the restoreState, it could call run on all the AllocRunner that it has, which would then call Run on all it's Task Runners? That way all the logic of how to run or attach to a Task is always encapsulated in the Run method. If the state of a Task is running the run method calls Open on a Driver otherwise it calls Start.